resolver-dns/src/main/java/io/netty/resolver/dns/DnsCache.java

```diff
@@ -45,24 +46,26 @@
     /**
      * Return the cached entries for the given hostname.
      * @param hostname the hostname
+     * @param additional the additional records
      * @return the cached entries
      */
-    List<DnsCacheEntry> get(String hostname);
+    List<DnsCacheEntry> get(String hostname, Iterable<DnsRecord> additional);
```

Comment:
I guess the intention of adding this parameter is to implement caching for the queries with additional records in the future, right? Given that we're going to have to convert this into a `List<DnsRecord>` or `DnsRecord[]` at least once, how about changing the parameter type to `DnsRecord[]` instead of `Iterable`?